src/System.Net.Primitives/tests/FunctionalTests/CredentialCacheTest.cs

```diff
@@ -162,7 +162,13 @@ public static void Add_HostPortAuthenticationTypeCredential_Invalid()
             Assert.Throws<ArgumentNullException>("host", () => cc.Add(null, 500, "authenticationType", new NetworkCredential())); //Null host
             Assert.Throws<ArgumentNullException>("authenticationType", () => cc.Add("host", 500, null, new NetworkCredential())); //Null authenticationType
 
-            Assert.Throws<ArgumentException>("host", () => cc.Add("", 500, "authenticationType", new NetworkCredential())); //Empty host
+            var exception = Record.Exception(() => cc.Add("", 500, "authenticationType", new NetworkCredential())); 
```

Comment:
Record.Exception will execute the Action code and will record the thrown exception and return it. if there is no exception thrown it will return null